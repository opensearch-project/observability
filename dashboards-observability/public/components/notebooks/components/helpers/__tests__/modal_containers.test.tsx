/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {
  DeleteNotebookModal,
  getCloneModal,
  getCustomModal,
  getDeleteModal,
  getSampleNotebooksModal,
} from '../modal_containers';

describe('modal_containers spec', () => {
  configure({ adapter: new Adapter() });

  it('render get custom modal', () => {
    const runModal = jest.fn();
    const closeModal = jest.fn();
    const wrapper = shallow(
      getCustomModal(
        runModal,
        closeModal,
        'label',
        'title',
        'Cancel',
        'Confirm',
        'mock-path',
        'help text'
      )
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render get clone modal', () => {
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const wrapper = shallow(getCloneModal(onCancel, onConfirm));
    expect(wrapper).toMatchSnapshot();
  });

  it('render get sample notebooks modal', () => {
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const wrapper = shallow(getSampleNotebooksModal(onCancel, onConfirm));
    expect(wrapper).toMatchSnapshot();
  });

  it('render get delete modal', () => {
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const wrapper = shallow(
      getDeleteModal(onCancel, onConfirm, 'Delete', 'Delete message', 'Confirm message')
    );
    expect(wrapper).toMatchSnapshot();

    const noConfirmMessageWrapper = shallow(
      getDeleteModal(onCancel, onConfirm, 'Delete', 'Delete message')
    );
    expect(noConfirmMessageWrapper).toMatchSnapshot();
  });

  it('render delete notebooks modal', () => {
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const wrapper = shallow(
      <DeleteNotebookModal
        onConfirm={onConfirm}
        onCancel={onCancel}
        title="title"
        message="message"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('checks delete notebooks modal', () => {
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const utils = render(
      <DeleteNotebookModal
        onConfirm={onConfirm}
        onCancel={onCancel}
        title="title"
        message="message"
      />
    );
    fireEvent.change(utils.getAllByTestId('delete-notebook-modal-input')[0], {
      target: { value: 'delete' },
    });
    fireEvent.click(utils.getAllByTestId('delete-notebook-modal-delete-button')[0]);
    expect(onConfirm).toBeCalled();
  });
});
