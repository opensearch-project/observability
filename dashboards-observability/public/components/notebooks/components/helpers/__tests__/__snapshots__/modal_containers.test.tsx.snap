// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal_containers spec render delete notebooks modal 1`] = `
<EuiOverlayMask>
  <EuiModal
    initialFocus="[name=input]"
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        title
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiText>
        message
      </EuiText>
      <EuiText>
        The action cannot be undone.
      </EuiText>
      <EuiSpacer />
      <EuiForm>
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          label="To confirm deletion, enter \\"delete\\" in the text field"
          labelType="label"
        >
          <EuiFieldText
            data-test-subj="delete-notebook-modal-input"
            name="input"
            onChange={[Function]}
            placeholder="delete"
            value=""
          />
        </EuiFormRow>
      </EuiForm>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiButtonEmpty
        onClick={[MockFunction]}
      >
        Cancel
      </EuiButtonEmpty>
      <EuiButton
        color="danger"
        data-test-subj="delete-notebook-modal-delete-button"
        disabled={true}
        fill={true}
        onClick={[Function]}
      >
        Delete
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;

exports[`modal_containers spec render get clone modal 1`] = `""`;

exports[`modal_containers spec render get custom modal 1`] = `
<EuiOverlayMask>
  <EuiModal
    initialFocus="[name=input]"
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        title
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiForm>
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          helpText="help text"
          label="label"
          labelType="label"
        >
          <EuiFieldText
            data-test-subj="custom-input-modal-input"
            name="input"
            onChange={[Function]}
            value="mock-path"
          />
        </EuiFormRow>
      </EuiForm>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiButtonEmpty
        onClick={[MockFunction]}
      >
        Cancel
      </EuiButtonEmpty>
      <EuiButton
        data-test-subj="custom-input-modal-confirm-button"
        fill={true}
        onClick={[Function]}
      >
        Confirm
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;

exports[`modal_containers spec render get delete modal 1`] = `""`;

exports[`modal_containers spec render get delete modal 2`] = `""`;

exports[`modal_containers spec render get sample notebooks modal 1`] = `""`;
